aws/database/.terraform.lock.hcl

```diff
@@ -0,0 +1,38 @@
+# This file is maintained automatically by "terraform init".
```

Comment:
Does it make sense to have this file here? I thought it's only useful for `terraform init`, but we don't run `terraform init` in a module.

Reply:
Yeah so this came up elsewhere, but basically, you can run terraform validate on a module, but it requires an init, hence this file. from an earlier mozilla/terraform-modules PR on this:

```
tf init & tf validate in module directory creates the .terraform.lock.hcl file... which i wasn't sure if we wanted to keep in modules specifically (it'll get ignored in usage in terraform projects as I understand it). this Hashi discussion led me to checking it in though
```

so I think we're good to commit these init lock files in modules where such CI may occur, but its not harmful to leave out

Reply:
this is the link: https://discuss.hashicorp.com/t/what-is-the-best-practise-with-terraform-lock-hcl-in-modules-commit-or-not/28648

Reply:
So this basically fixes the versions of dependencies that are used when running `terraform validate`. Including the file here has the advantage of having more reproducible builds, but the downside that we need to update the file every now and then, and that we might catch some errors later.

I don't really have an opinion either way, so let's just keep it here.